filesystems/filesystem/src/main/java/org/angellee/CharsetDetector.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019-2020 gzu-liyujiang <1032694760@qq.com>
+ *
+ * The software is licensed under the Mulan PSL v1.
+ * You can use this software according to the terms and conditions of the Mulan PSL v1.
+ * You may obtain a copy of Mulan PSL v1 at:
+ *     http://license.coscl.org.cn/MulanPSL
+ * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR
+ * PURPOSE.
+ * See the Mulan PSL v1 for more details.
+ *
+ */
+
+package org.angellee;
+
+import java.io.ByteArrayInputStream;
+
+/**
+ * Adapted from https://github.com/angelLYK/smartCharsetDetector
+ */
+public interface CharsetDetector {
```

Comment:
Please use Kotlin everywhere, it's the main language for this repo. Kotlin is not that hard for Java devs anyway